pkg/flare/archive.go

```diff
@@ -148,11 +150,13 @@ func provideSystemProbe(fb flaretypes.FlareBuilder) error {
 	addSystemProbePlatformSpecificEntries(fb)
 
 	if pkgconfigsetup.SystemProbe().GetBool("system_probe_config.enabled") {
-		fb.AddFileFromFunc(filepath.Join("expvar", "system-probe"), getSystemProbeStats)                         //nolint:errcheck
-		fb.AddFileFromFunc(filepath.Join("system-probe", "system_probe_telemetry.log"), getSystemProbeTelemetry) // nolint:errcheck
-		fb.AddFileFromFunc(filepath.Join("system-probe", "conntrack_cached.log"), getSystemProbeConntrackCached) // nolint:errcheck
-		fb.AddFileFromFunc(filepath.Join("system-probe", "conntrack_host.log"), getSystemProbeConntrackHost)     // nolint:errcheck
-		fb.AddFileFromFunc(filepath.Join("system-probe", "ebpf_btf_loader.log"), getSystemProbeBTFLoaderInfo)    // nolint:errcheck
+		_ = fb.AddFileFromFunc(filepath.Join("expvar", "system-probe"), getSystemProbeStats)
+		_ = fb.AddFileFromFunc(filepath.Join("system-probe", "system_probe_telemetry.log"), getSystemProbeTelemetry)
+		if runtime.GOOS == "linux" {
```

Comment:
It seems that we are doing the check on the platform using the separate files `archive_no_linux` and `archive_linux`. 

If we are going to use the  `runtime.GOOS == "linux"` branch I would suggest to remove specfic files in `flare/systemprobe`

Reply:
I did this because those functions return an error (which gets logged) on unsupported platforms. I could:

1. Change them to not return an error on unsupported platforms, but that would result in empty files in the flare.
2. Refactor the code to not have `pkg/flare/sysprobe/archive_nolinux.go` and `pkg/flare/sysprobe/archive_unsupported.go`.

Do you have a preference?

Reply:
I reworked this, taking advantage of some functions that already existed. Let me know what you think.

Reply:
The one downside to that, is that anything common to all platforms would get repeated.

Reply:
Good point. Less leave like this. As state on another comment https://github.com/DataDog/datadog-agent/pull/30936#discussion_r1840644407 this could be remove in a later PR to flare providers 😄